src/modules/Home/selectors/home.ts

```diff
@@ -0,0 +1,16 @@
+import {createSelector} from 'reselect';
+
+export const getMainSlides = (state: any) => state.home.mainSlides;
+export const getCurrentMainSlideIndex = (state: any) => state.home.currentSlideIndex;
+export const getIsMainSliderAnimated = (state: any) => state.home.isMainSliderAnimated;
+
+export const getMainSlidesLength = createSelector(
+	getMainSlides,
+	slides => slides.length,
+);
+
+export const getCurrentMainSlide = createSelector(
+	getMainSlides,
+	getCurrentMainSlideIndex,
+	(slides, current) => slides[current],
```

Comment:
the complexity of this algorithm is O1. state => selector1[selecor2] is enough

Reply:
Fixed